server/internal/infrastructure/gcp/cloudtasks.go

```diff
@@ -2,46 +2,90 @@ package gcp
 
 import (
 	"context"
+	"encoding/json"
 
+	cloudtasks "cloud.google.com/go/cloudtasks/apiv2"
 	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
 	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/id/idx"
 	"github.com/reearth/reearth-cms/server/pkg/task"
+	"google.golang.org/api/option"
+	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2"
 )
 
-// Queue tasks care of queuing and Tasks will be added to it
-type Queue struct {
+type TaskRunner struct {
+	queuePath     string
+	subscriberURL string
+	credFilePath  string
 }
 
-type RequestType string
+func NewTaskRunner(c *CloudTasksConfig, opts ...TaskRunnerOption) (gateway.TaskRunner, error) {
+	opts2 := defaultTaskRunnerOptions()
+	for _, o := range opts {
+		o.Apply(opts2)
+	}
 
-const (
-	HTTP = RequestType("http")
-)
+	qURL, err := c.buildQueueUrl()
+	if err != nil {
+		return nil, err
+	}
 
-// Task is request for subscriber
-type Task struct {
-	reqType RequestType
-	payload any
+	return &TaskRunner{
+		queuePath:     qURL,
+		subscriberURL: c.SubscriberURL,
+		credFilePath:  opts2.credFilePath,
+	}, nil
 }
 
-type Config struct {
-	GCPProject string
-	GCPRegion  string
-	QueueName  string
-}
+// Run implements gateway.TaskRunner
+func (t *TaskRunner) Run(ctx context.Context, p task.Payload) (id.TaskID, error) {
 
-func NewTaskRunner(c *Config) gateway.TaskRunner {
-	// TODO: convert config to Queue struct
-	return &Queue{}
-}
+	client, closeFn, err := t.client(ctx)
+	if err != nil {
+		return idx.ID[id.Task]{}, err
+	}
+	defer closeFn()
+
+	bPayload, err := json.Marshal(p.DecompressAsset.Payload())
+	if err != nil {
+		return idx.ID[id.Task]{}, err
+	}
+	req := t.buildRequest(t.subscriberURL, bPayload)
+
+	_, err = client.CreateTask(ctx, req)
+	if err != nil {
+		return idx.ID[id.Task]{}, err
+	}
 
-func NewTask() *Task {
 	//TODO: impl here
-	return &Task{}
+	return idx.ID[id.Task]{}, nil
```

Comment:
Should I return task's id though CloudTasks doesn't. If that's the case, need to generate Task's id which won't be used anymore.